Corona-Warn-App/src/test/java/de/rki/coronawarnapp/ui/launcher/LauncherActivityViewModelTest.kt

```diff
@@ -249,6 +251,47 @@ class LauncherActivityViewModelTest : BaseTest() {
         }
     }
 
+    @Test
+    fun `device not rooted triggers update check`() {
+        coEvery { rootDetectionCheck.isRooted() } returns false
+
+        coEvery { updateChecker.checkForUpdate() } returns UpdateChecker.Result(isUpdateNeeded = true)
+        coEvery { appUpdateManager.getUpdateInfo() } returns
+            mockk<AppUpdateInfo>().apply {
+                every { updateAvailability() } returns UpdateAvailability.UPDATE_AVAILABLE
+            }
+
+        createViewModel().run {
+            onRootedDialogDismiss()
+            events.getOrAwaitValue() shouldBe instanceOf(LauncherEvent.ShowUpdateDialog::class)
+        }
+
+        coVerify {
+            rootDetectionCheck.isRooted()
+        }
+    }
+
+    @Test
+    fun `rooted device but suppress root dialog`() {
+        coEvery { rootDetectionCheck.isRooted() } returns true
+        coEvery { rootDetectionCheck.shouldShowRootInfo } returns false
+
+        coEvery { updateChecker.checkForUpdate() } returns UpdateChecker.Result(isUpdateNeeded = true)
+        coEvery { appUpdateManager.getUpdateInfo() } returns
+            mockk<AppUpdateInfo>().apply {
+                every { updateAvailability() } returns UpdateAvailability.UPDATE_AVAILABLE
+            }
+
+        createViewModel().run {
+            onRootedDialogDismiss()
```

Comment:
I think we shouldn't call this method, as it also won't be called in production in this case. The test case is still successful after deleting it.

Reply:
dne